yargs.js

```diff
@@ -241,7 +249,7 @@ function Yargs (processArgs, cwd, parentRequire) {
 
     if (typeof keys === 'number') {
       if (!options.demanded._) options.demanded._ = { count: 0, msg: null, max: max }
-      options.demanded._.count = keys
+      options.demanded._.count = keys // + context.commands.length
```

Comment:
The comment is meant to show what yargs should possibly be doing to make `.demand(num)` contextual. This by itself may not be sufficient though (e.g. error message handling). I can remove it from this PR.